src/LinuxTracing/LeafFunctionCallManagerTest.cpp

```diff
@@ -149,15 +149,16 @@ class LeafFunctionCallManagerTest : public ::testing::Test {
 
 CallchainSamplePerfEventData BuildFakeCallchainSamplePerfEventData(
     const std::vector<uint64_t>& callchain) {
-  CallchainSamplePerfEventData event_data{
-      .pid = 10,
-      .tid = 11,
-      .regs = make_unique_for_overwrite<perf_event_sample_regs_user_all>(),
-      .data = make_unique_for_overwrite<uint8_t[]>(13)};
+  CallchainSamplePerfEventData event_data{.pid = 10,
+                                          .tid = 11,
+                                          .regs = make_unique_for_overwrite<uint64_t[]>(20),
+                                          .data = make_unique_for_overwrite<uint8_t[]>(13)};
+
   event_data.SetIps(callchain);
+
   if (callchain.size() > 1) {
     // Set the first non-kernel address as IP.
-    event_data.regs->ip = callchain[1];
+    event_data.regs.get()[8] = callchain[1];
```

Comment:
```suggestion
    perf_event_sample_regs_user_all regs{};
    regs.ip = callchain[1];
    event_data.regs = std::memcpy(event.regs.get(), &regs, sizeof(regs));
```

Reply:
The last line should just be `std::memcpy(event_data.regs.get(), &regs, sizeof(regs));`, right?

Reply:
exactly.

Reply:
Done.